### ISSUE TITLE
Header is not accessible with VoiceOver in IOS Simulator when there is a headerTitle, headerLeft, and headerRight

### ISSUE THREAD

**Issue by lgibso34** [react-navigation/react-navigation#11387]
### Current behavior

The header for the stack navigator is not accessible (skipped over) by Mac VoiceOver (VO) when there is a `headerTitle`, `headerLeft`, and a `headerRight`. A physical iPhone VOI can read these fine.

However, if you comment out [this line](https://github.com/lgibso34/expo-test-app/blob/master/RootNavigator.tsx#L20) from my test repo, you'll notice that VO can now traverse into the navigation bar and read the `headerLeft` content and the header title. So, you can remove either the `headerRight` or `headerLeft` and VO will now be able to reach the content in the header area. It is specifically when both are present when VO can't make it into the header.

I looked into the `react-navigation` code and couldn't find anything obvious that would cause this.

### Expected behavior

VO on a iOS simulator should be able to read text within a header when there is a `headerTitle`, `headerLeft`, and a `headerRight` for a Stack.

Once again, VO works as expected on a physical device.

### Reproduction

https://github.com/lgibso34/expo-test-app

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |^6.1.6
| @react-navigation/native-stack         |^6.9.12
| react-native                           |0.71.8
| expo                                   |48.0.15
| node                                   |18
| npm or yarn                            |9.5.1
